src/plugins/data/public/search/aggs/buckets/geo_hash_fn.ts

```diff
@@ -0,0 +1,128 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { Assign } from '@kbn/utility-types';
+import { ExpressionFunctionDefinition } from '../../../../../expressions/public';
+import { AggExpressionType, AggExpressionFunctionArgs, BUCKET_TYPES } from '../';
+import { getParsedValue } from '../utils/get_parsed_value';
+
+const fnName = 'aggGeoHash';
+
+type Input = any;
+type AggArgs = AggExpressionFunctionArgs<typeof BUCKET_TYPES.GEOHASH_GRID>;
+
+type Arguments = Assign<AggArgs, { boundingBox?: string }>;
+type Output = AggExpressionType;
+type FunctionDefinition = ExpressionFunctionDefinition<typeof fnName, Input, Arguments, Output>;
+
+export const aggGeoHash = (): FunctionDefinition => ({
+  name: fnName,
+  help: i18n.translate('data.search.aggs.function.buckets.geoHash.help', {
+    defaultMessage: 'Generates a serialized agg config for a Geo Hash agg',
+  }),
+  type: 'agg_type',
+  args: {
+    id: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.geoHash.id.help', {
+        defaultMessage: 'ID for this aggregation',
+      }),
+    },
+    enabled: {
+      types: ['boolean'],
+      default: true,
+      help: i18n.translate('data.search.aggs.buckets.geoHash.enabled.help', {
+        defaultMessage: 'Specifies whether this aggregation should be enabled',
+      }),
+    },
+    schema: {
+      types: ['string'],
+      help: i18n.translate('data.search.aggs.buckets.geoHash.schema.help', {
+        defaultMessage: 'Schema to use for this aggregation',
+      }),
+    },
+    field: {
+      types: ['string'],
+      required: true,
+      help: i18n.translate('data.search.aggs.buckets.geoHash.field.help', {
+        defaultMessage: 'Field to use for this aggregation',
+      }),
+    },
+    useGeocentroid: {
+      types: ['boolean'],
+      required: false,
+      help: i18n.translate('data.search.aggs.buckets.geoHash.useGeocentroid.help', {
+        defaultMessage: 'Should be used geocentroid for this aggregation or not?',
+      }),
+    },
+    autoPrecision: {
+      types: ['boolean'],
+      required: false,
+      help: i18n.translate('data.search.aggs.buckets.geoHash.autoPrecision.help', {
+        defaultMessage: 'Should be used auto precision for this aggregation or not?',
```

Comment:
It might be nice to make the wording on each of the boolean messages similar to the one for `enabled` above, i.e.

```suggestion
        defaultMessage: 'Specifies whether to use auto precision for this aggregation',
```